src/datasets/arrow_dataset.py

```diff
@@ -616,6 +616,27 @@ def _align_features(features_list: List[Features]) -> List[Features]:
     return [Features({k: name2feature[k] for k in features.keys()}) for features in features_list]
 
 
+def _check_if_trying_to_flatten_audio_feature(feature, column_name, depth, max_depth, inplace: bool = False):
+    """Check if the user is trying to flatten an audio feature."""
+    if isinstance(feature, Audio) and feature.decode:
```

Comment:
Why audio and not image as well here ?

Reply:
A bit hard to explain. We can't dispatch single storage to multiple columns that are introduced by flatten ("path", "array", "sampling_rate" for Audio). And for `Image` we can, because `flatten` is a no-op in that case (singe storage -> single column).